Tests/src/PublicAPITests/HotspotSequenceTests.cpp

```diff
@@ -689,4 +692,400 @@ CSP_PUBLIC_TEST(CSPEngine, HotspotSequenceTests, GenerateSequenceKeyTest)
+
+	auto [EnterResult] = AWAIT_PRE(SpaceSystem, EnterSpace, RequestPredicate, Space.Id);
+
+	EXPECT_EQ(EnterResult.GetResultCode(), csp::systems::EResultCode::Success);
+
+	EntitySystem->SetEntityCreatedCallback(
+		[](csp::multiplayer::SpaceEntity* Entity)
+		{
+		});
+
+	// Create object to represent the hotspot
+	csp::common::String ObjectName					 = "Object 1";
+	csp::multiplayer::SpaceTransform ObjectTransform = {csp::common::Vector3::Zero(), csp::common::Vector4::Zero(), csp::common::Vector3::One()};
+	auto [CreatedObject]							 = AWAIT(EntitySystem, CreateObject, ObjectName, ObjectTransform);
+
+	bool ComponentAdded = false;
+
+	CreatedObject->SetUpdateCallback(
+		[&ComponentAdded, ObjectName](csp::multiplayer::SpaceEntity* Entity,
+									  csp::multiplayer::SpaceEntityUpdateFlags Flags,
+									  csp::common::Array<csp::multiplayer::ComponentUpdateInfo>& UpdateInfo)
+		{
+			if (Entity->GetName() == ObjectName)
+			{
+				for (size_t i = 0; i < UpdateInfo.Size(); ++i)
+				{
+					if (UpdateInfo[i].UpdateType == csp::multiplayer::ComponentUpdateType::Add)
+					{
+						ComponentAdded = true;
+					}
+				}
+			}
+		});
+
+	// Create hotspot component
+	auto* HotspotComponent
+		= static_cast<csp::multiplayer::HotspotSpaceComponent*>(CreatedObject->AddComponent(csp::multiplayer::ComponentType::Hotspot));
+
+	CreatedObject->QueueUpdate();
+	WaitForCallbackWithUpdate(ComponentAdded, EntitySystem);
+
+	EXPECT_TRUE(ComponentAdded);
+
+	// Create Hotspot groups
+	csp::common::String TestGroupName1 = "CSP-UNITTEST-SEQUENCE-MAG1";
+	csp::common::String TestGroupName2 = "CSP-UNITTEST-SEQUENCE-MAG2";
+	csp::common::String TestGroupName3 = "CSP-UNITTEST-SEQUENCE-MAG3";
+
+	{
+		// Create 2 groups that contains the component
+		csp::systems::HotspotGroup HotspotGroup1;
+		CreateHotspotgroup(HotspotSystem, TestGroupName1, {HotspotComponent->GetUniqueComponentId()}, HotspotGroup1);
+
+		csp::systems::HotspotGroup HotspotGroup2;
+		CreateHotspotgroup(HotspotSystem, TestGroupName2, {HotspotComponent->GetUniqueComponentId()}, HotspotGroup2);
+
+		// Create another group that doesnt contain the component
+		csp::systems::HotspotGroup HotspotGroup3;
+		CreateHotspotgroup(HotspotSystem, TestGroupName3, {"TestName"}, HotspotGroup3);
+
+		// Ensure the 3 groups are created correctly
+		csp::common::Array<csp::systems::HotspotGroup> FoundGroups;
+		GetHotspotGroups(HotspotSystem, {TestGroupName1, TestGroupName2, TestGroupName3}, FoundGroups);
+
+		EXPECT_EQ(FoundGroups.Size(), 3);
+	}
+
+	// Exit the space
+	auto [ExitSpaceResult] = AWAIT_PRE(SpaceSystem, ExitSpace, RequestPredicate);
+	// Reenter the space
+	auto [ReEnterResult] = AWAIT_PRE(SpaceSystem, EnterSpace, RequestPredicate, Space.Id);
+
+	// Ensure the 3 groups still exist
+	csp::common::Array<csp::systems::HotspotGroup> FoundGroups;
+	GetHotspotGroups(HotspotSystem, {TestGroupName1, TestGroupName2, TestGroupName3}, FoundGroups);
```

Comment:
Do you actually need to be in a space to get hotspot groups that relate to the space? If not, I'm not sure if this test is necessary?

Reply:
HotspotSequence api relies on people being in a space